### PR TITLE
Feature/1600 payment recursion

### LINKED ISSUE #1600: In some cases, changing an Opportunity to Closed/Lost causes infinite recursion
https://powerofus.force.com/_ui/core/userprofile/UserProfilePage?u=00580000005WuCJ&tab=sfdc.ProfilePlatformFeed&fId=0D580000028Qlbg&emtm=1433784041080&s1nid=0DB800000004Fa7&OpenCommentForEdit=1&fromEmail=1&emkind=chatterCommentNotification&s1ext=0&s1oid=00D00000000hWLM&s1uid=00580000006ktn4

### LINKED ISSUE #1605: Address Verification Service drop down not giving choice of services
POU thread: https://powerofus.force.com/0D580000029W6KB

Reproduced in org with 3.42 and when editing Address Settings, the Services drop down does not show any services to select.

### PR DESCRIPTION
**Pull request by njjc**
# Warning
# Info
# Issues
Fixes #1600
Fixes #1605

### CODE REVIEW

File: src/classes/PMT_PaymentCreator.cls [hunk 1]
```diff
@@ -163,16 +164,9 @@ public without sharing class PMT_PaymentCreator {
                         }
                         */
                 }
-                try{
```
Comment (njjc) [inline src/classes/PMT_PaymentCreator.cls]: This try would never catch anything as we're no longer doing the DML here.

File: src/classes/PMT_PaymentCreator.cls [hunk 2]
```diff
@@ -263,20 +257,13 @@ public without sharing class PMT_PaymentCreator {
                         paymentsToInsert.add(op);                                              
                     }   
                 }
-                //try{
```
Comment (njjc) [inline src/classes/PMT_PaymentCreator.cls]: Trying to remove old commented out code that we don't need anymore when I see it in a file I'm working on.

File: src/classes/PMT_PaymentCreator_TEST.cls [hunk 3]
```diff
@@ -131,5 +133,44 @@ private with sharing class PMT_PaymentCreator_TEST {
...
+            Amount = 1,
+            StageName = UTIL_UnitTestData_TEST.getOpenStage()
+        );
+        insert opp;
+
+        list<npe01__OppPayment__c> opmtAll = [select id, npe01__payment_method__c, npe01__payment_amount__c, npe01__paid__c, npe01__written_off__c from npe01__OppPayment__c WHERE npe01__opportunity__c = :opp.Id];
+        system.assertEquals(1, opmtAll.size());
+
+        opp.Amount = null;
+        opp.StageName = 'Closed Lost';
```
Comment (njjc) [inline src/classes/PMT_PaymentCreator_TEST.cls]: We have a "getClosedLostStage()" method in UTIL_UnitTestData_TEST that's commented out. I would have used it here if we had it; do you think I should add it back?

**Comment by njjc** [comment]
@davidhabib ready for review and merge pending build. Might also want to wait until the last merge to dev's build status is fixed.

File: src/classes/PMT_PaymentCreator_TEST.cls (hunk 3 above)
Comment (davidhabib) [inline src/classes/PMT_PaymentCreator_TEST.cls]: yes, let's add it back.

**Comment by njjc** [comment]
@davidhabib good to go

**Pull request merged**